src/OrbitGl/App.h

```diff
@@ -289,11 +289,20 @@ class OrbitApp final : public DataViewFactory, public CaptureListener {
       const std::vector<ModuleData*>& modules,
       absl::flat_hash_map<std::string, std::vector<uint64_t>> function_hashes_to_hook_map,
       absl::flat_hash_map<std::string, std::vector<uint64_t>> frame_track_function_hashes_map);
-  orbit_base::Future<ErrorMessageOr<void>> LoadModule(const ModuleData* module);
-  orbit_base::Future<ErrorMessageOr<void>> LoadModule(const std::string& module_path,
-                                                      const std::string& build_id);
+
+  // LoadModule retrieves a module file and returns the local file path (potentially from the local
+  // cache). Symbols won't be loaded.
+  orbit_base::Future<ErrorMessageOr<std::filesystem::path>> LoadModule(const ModuleData* module);
+  orbit_base::Future<ErrorMessageOr<std::filesystem::path>> LoadModule(
+      const std::string& module_path, const std::string& build_id);
   orbit_base::Future<void> LoadModules(absl::Span<const ModuleData* const> modules);
 
+  // LoadModuleAndSymbols is a helper function which first retrieves the module by calling
+  // `LoadModule` and afterwards load the symbols by calling `LoadSymbols`.
+  orbit_base::Future<ErrorMessageOr<void>> LoadModuleAndSymbols(const ModuleData* module);
+  orbit_base::Future<ErrorMessageOr<void>> LoadModuleAndSymbols(const std::string& module_path,
+                                                                const std::string& build_id);
```

Comment:
Is this callback making sense? It is only called from `LoadModuleAndSymbols(const ModuleData* module)`. I am mainly curious whether you designed it like this on purpose

Reply:
I guess you mean overload, do you? Why do you think it doesn't make sense? You can call it whenever you don't have a ModuleData object. You actually don't need a full ModuleData object, a path and a build_id is fine.

Reply:
Yes I meant overload instead of callback. 
> You actually don't need a full ModuleData object, a path and a build_id is fine.

But there currently does not seem to be a scenario where only path and build_id is available, but not the full ModuleData object. Are you anticipating there will be a scenario in the future, or is there any other reason? 

I could imagine that this is good design in the sense of that `LoadModuleAndSymbols(path, build_id)` is the minimal function that is maybe more easily testable or other better for other reasons? And `LoadModuleAndSymbols(module_data)` is the convenience overload. Is this your thinking behind it, or something else?

Reply:
> Yes I meant overload instead of callback.
> 
> > You actually don't need a full ModuleData object, a path and a build_id is fine.
> 
> But there currently does not seem to be a scenario where only path and build_id is available, but not the full ModuleData object. Are you anticipating there will be a scenario in the future, or is there any other reason?
> 
> I could imagine that this is good design in the sense of that `LoadModuleAndSymbols(path, build_id)` is the minimal function that is maybe more easily testable or other better for other reasons? And `LoadModuleAndSymbols(module_data)` is the convenience overload. Is this your thinking behind it, or something else?

Yes, that's exactly my reasoning here. I try to explicitly state the dependencies and keep them minimal. That helps with decoupling in the long run as well. Do you disagree?